pgtrigger/core.py

```diff
@@ -523,9 +525,24 @@ def render(self, model: models.Model) -> str:
         Returns:
             The rendered SQL.
         """
-        fields = utils.AttrDict({field.name: field for field in model._meta.fields})
-        columns = utils.AttrDict({field.name: field.column for field in model._meta.fields})
-        return self.func.format(meta=model._meta, fields=fields, columns=columns)
+        kwargs = {
+            "meta": model._meta,
+            "fields": utils.AttrDict({field.name: field for field in model._meta.fields}),
+            "columns": utils.AttrDict({field.name: field.column for field in model._meta.fields}),
+            "reset_ignore": (
+                """
+                IF _prev_ignore IS NOT NULL AND (_prev_ignore = '') IS NOT TRUE THEN
+                    PERFORM set_config('pgtrigger.ignore', _prev_ignore, true);
+                ELSE
+                    PERFORM set_config('pgtrigger.ignore', '', true);
+                END IF;
+                """
+                if trigger.ignores_others
+                else ""
+            ),
+        } | self.kwargs
+
+        return self.func.format(**kwargs)
```

Comment:
`Func` doesn't seem to be the optimal place for generating boilerplate wrapper SQL like this. For example, [this](https://github.com/Opus10/django-pgtrigger/blob/a3728c59167d766c99ce4c7e80389903af3b9f29/pgtrigger/compiler.py#L67) is where triggers are wrapped with the ability to be ignored. I assumed that we would also generate this boilerplate in a similar way. Having a `self.kwargs` variable also strikes me as a hack.

Was there a reason behind overriding `Func`?

Reply:
Only that Func seemed to be only a wrapper for string formatting. This extends it to also provide custom context. See docs/ignoring_triggers.md:101

Reply:
About your example in `compiler.py`. I could not think of a way to wrap a user defined trigger body in such a way that resetting the _pgtrigger_ignore would always get executed. The user might return from any branch in their script.
The user would have to decide when to reset it